tools/version-bump/lib/update.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * External dependencies
+ */
+import { readFile, writeFile, stat } from 'fs/promises';
+
+/**
+ * Internal dependencies
+ */
+import { Logger } from './logger';
+
+/**
+ * Update plugin readme stable tag.
+ *
+ * @param  plugin      plugin to update
+ * @param  nextVersion version to bump to
+ */
+export const updateReadmeStableTag = async (
+	plugin: string,
+	nextVersion: string
+): Promise< void > => {
+	const filePath = `plugins/${ plugin }/readme.txt`;
+	try {
+		const readmeContents = await readFile( filePath, 'utf8' );
+
+		const updatedReadmeContents = readmeContents.replace(
+			/Stable tag: \d.\d.\d\n/m,
+			`Stable tag: ${ nextVersion }\n`
+		);
+
+		await writeFile( filePath, updatedReadmeContents );
+	} catch ( e ) {
+		Logger.error( 'Unable to update readme stable tag' );
+	}
+};
+
+/**
+ * Update plugin readme changelog.
+ *
+ * @param  plugin      plugin to update
+ * @param  nextVersion version to bump to
+ */
+export const updateReadmeChangelog = async (
+	plugin: string,
+	nextVersion: string
+): Promise< void > => {
+	const filePath = `plugins/${ plugin }/readme.txt`;
+	try {
+		const readmeContents = await readFile( filePath, 'utf8' );
+
+		const updatedReadmeContents = readmeContents.replace(
+			/= \d.\d.\d \d\d\d\d-XX-XX =\n/m,
+			`= ${ nextVersion } ${ new Date().getFullYear() }-XX-XX =\n`
+		);
+
+		await writeFile( filePath, updatedReadmeContents );
+	} catch ( e ) {
+		Logger.error( 'Unable to update readme changelog' );
+	}
+};
+
+/**
+ * Update plugin class file.
+ *
+ * @param  plugin      plugin to update
+ * @param  nextVersion version to bump to
+ */
+export const updateClassPluginFile = async (
+	plugin: string,
+	nextVersion: string
+): Promise< void > => {
```

Comment:
Just a heads up, TypeScript is very good at inferring the return type so unless you want to make it explicit what's being returned you don't need to add the return type.

Reply:
Ok, good to know. Since its in there, I'll leave it because it makes it apparent.